rlclientlib/multistep.cc

```diff
@@ -0,0 +1,44 @@
+#include "multistep.h"
+#include "err_constants.h"
+
+namespace reinforcement_learning {
+  void episode_history::update(const char* event_id, const char* previous_event_id, const char* context, const ranking_response& resp) {
+    _depths[event_id] = this->get_depth(previous_event_id) + 1;
+  }
+
+  std::string episode_history::get_context(const char* previous_event_id, const char* context) const {
+    return R"({"episode":{"depth":")" + std::to_string(this->get_depth(previous_event_id) + 1) + "\"}," + std::string(context + 1);
```

Comment:
1. Probably better to discuss with broader audience whether it is ok for now to introduce hard-coded namespace.
2. If so, make json manipulations cleaner

Reply:
yes agreed.

Reply:
More detailed explanation of the problems with this piece of code:
1) episode history class is supposed to be splitted into interface (which is introducing extension point) and implementation (existing one which is tracking #step is just one of possible implementations).
For existing implementation:
1) Injecting new feature into hardcoded namespace/feature_name is definitely a hack (obvious problem is existence of the same namespace in the context).
2) json manipulation is clearly bad.

Given that this code is still considered as prototype and not interfering with any of cb/ccb stack, it is fine to me to commit as is and work on improvements incrementally, but can you please share your thoughts (@olgavrou @jackgerrits @lokitoth )?

Reply:
Since it doesn't affect any path that is in use I think it's fine to make changes and cleanups incrementally